frontend/src/routes/courses/+page.svelte

```diff
@@ -0,0 +1,93 @@
+<script lang="ts">
+  import { onMount } from 'svelte';
+  import { writable } from 'svelte/store';
+  import { get } from '$utils/fetch';
+  import { _ } from 'svelte-i18n';
+
+  interface Course {
+    name: string;
+    user: {
+      first_name: string;
+    };
+  }
+
+  let courses = writable<Course[]>([]);
+  let search = '';
+
+  async function submitForm() {
+    const query = encodeURIComponent(search.trim());
+    window.location.href = `/courses?search=${query}`;
+  }
+
+  async function getCourses() {
+    const urlSearchParams = new URLSearchParams(window.location.search);
+    search = urlSearchParams.get('search') || '';
```

Comment:
You could get this data by using sveltekit parameters: https://kit.svelte.dev/docs/types#public-types-page